import: remove deprecated fields from ImportDetails proto

yuzefovich opened the pull request:
**import: remove deprecated fields from ImportDetails proto**

The following no longer used fields are removed from the ImportDetails proto:
- SSTSize: this option stopped having any effect in 8790323d2d8bd6daff61cdb1566e04a38791d252. This WITH option of the IMPORT stmt was already skipped from documentation (or was never documented).
- Oversample
- BackupPath
- Samples
- IngestDirectly
- ProtectedTimestampRecord: this was deprecated in 4c9455048921eeb0d38af00e011cbfba321e1fc6.

**import: address some nits**

This commit addresses a few nits (like typos in the comments and removing a no longer used method from an interface).

`SSTBatcher.initialSplitDone` is unused as of e4e41efda5854763a5f90626bec633ec2ce3f57e.

Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/146906)
<!-- Reviewable:end -->

yuzefovich committed:
import: remove deprecated fields from ImportDetails proto

The following no longer used fields are removed from the ImportDetails
proto:
- SSTSize: this option stopped having any effect in
8790323d2d8bd6daff61cdb1566e04a38791d252. This WITH option of the IMPORT
stmt was already skipped from documentation (or was never documented).
- Oversample
- BackupPath
- Samples
- IngestDirectly
- ProtectedTimestampRecord: this was deprecated in 4c9455048921eeb0d38af00e011cbfba321e1fc6.

Release note: None

yuzefovich committed:
import: address some nits

This commit addresses a few nits (like typos in the comments and
removing a no longer used method from an interface).

`SSTBatcher.initialSplitDone` is unused as of
e4e41efda5854763a5f90626bec633ec2ce3f57e.

Release note: None

michae2 reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Thanks for doing this!

> Reviewed 13 of 13 files at r1, 13 of 13 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/146906#-:-OQd_Trr932o6zxKJ87b:b-pt8m11)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @yuzefovich)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+
